rust/worker/src/execution/functions/statistics.rs

```diff
@@ -0,0 +1,786 @@
+    fn stable_value(&self) -> String {
+        // NOTE(rescrv):  Keep in sync with stable_string.  Done separately to avoid clone.
+        match self {
+            Self::Bool(b) => {
+                format!("{b}")
+            }
+            Self::Int(i) => {
+                format!("{i}")
+            }
+            Self::Str(s) => s.clone(),
+            Self::Float(f) => format!("{f:.16e}"),
+            Self::SparseVector(index) => {
+                format!("{index}")
+            }
+        }
+    }
+
+    /// A stable string representation of a statistics value with type tag.
+    /// Separate so display repr can change.
+    fn stable_string(&self) -> String {
+        // NOTE(rescrv):  Keep in sync with stable_value.  Done separately to avoid clone.
+        match self {
+            Self::Bool(b) => {
+                format!("b:{b}")
+            }
+            Self::Int(i) => {
+                format!("i:{i}")
+            }
+            Self::Str(s) => {
+                format!("s:{s}")
+            }
+            Self::Float(f) => format!("f:{f:.16e}"),
+            Self::SparseVector(index) => {
+                format!("sv:{index}")
+            }
+        }
+    }
+}
+
+impl std::fmt::Display for StatisticsValue {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.stable_string())
+    }
+}
+
+impl PartialEq for StatisticsValue {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Bool(lhs), Self::Bool(rhs)) => lhs == rhs,
+            (Self::Int(lhs), Self::Int(rhs)) => lhs == rhs,
+            (Self::Float(lhs), Self::Float(rhs)) => lhs.to_bits() == rhs.to_bits(),
+            (Self::Str(lhs), Self::Str(rhs)) => lhs == rhs,
+            (Self::SparseVector(lhs), Self::SparseVector(rhs)) => lhs == rhs,
+            _ => false,
+        }
+    }
+}
+
+impl Eq for StatisticsValue {}
+
+impl Hash for StatisticsValue {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        std::mem::discriminant(self).hash(state);
+        match self {
+            StatisticsValue::Bool(value) => value.hash(state),
+            StatisticsValue::Int(value) => value.hash(state),
+            StatisticsValue::Float(value) => value.to_bits().hash(state),
+            StatisticsValue::Str(value) => value.hash(state),
+            StatisticsValue::SparseVector(value) => value.hash(state),
+        }
+    }
+}
+
+/// Task executor that aggregates metadata value frequencies for the statistics task.
+#[derive(Debug)]
+pub struct StatisticsFunctionExecutor;
+
+#[async_trait]
+impl TaskExecutor for StatisticsFunctionExecutor {
+    async fn execute(
+        &self,
+        input_records: Chunk<LogRecord>,
+        output_reader: Option<&RecordSegmentReader<'_>>,
+    ) -> Result<Chunk<LogRecord>, Box<dyn ChromaError>> {
+        let mut counts: HashMap<(String, StatisticsValue), i64> = HashMap::default();
```

Comment:
Suggestion: What if we had a map from metadata key to a Statistics object that was able to compute multiple statistics types, not just count? Say we potentially wanted averages or ranges for the integer keys.